pavelib/paver_tests/test_database.py

```diff
@@ -59,3 +64,141 @@ def test_extract_files_from_zip(self, _mock_verify):
+        the stored fingerprint, verify that we make a call to load data into
+        the database without running migrations
+        """
+        self.addCleanup(shutil.rmtree, db_utils.CACHE_FOLDER)
+        self.addCleanup(os.remove, db_utils.FINGERPRINT_FILEPATH)
+        _write_temporary_db_cache_files(db_utils.CACHE_FOLDER, database.ALL_DB_FILES)
+        # write the local fingerprint file with the same value than the
+        # computed fingerprint
+        expected_fingerprint = 'ccaa8d8dcc7d030cd6a6768db81f90d0ef976c3d'
+        with open(db_utils.FINGERPRINT_FILEPATH, 'w') as fingerprint_file:
+            fingerprint_file.write(expected_fingerprint)
+
+        database.update_local_bokchoy_db_from_s3()
+        calls = [
+            call('{}/scripts/calculate-bokchoy-migrations.sh'.format(Env.REPO_ROOT)),
+            call('{}/scripts/reset-test-db.sh'.format(Env.REPO_ROOT))
+        ]
+        _mock_sh.assert_has_calls(calls)
+
+    @patch.object(database, 'CACHE_BUCKET_NAME', 'moto_test_bucket')
+    @patch.object(db_utils, 'CACHE_FOLDER', mkdtemp())
+    @patch.object(db_utils, 'FINGERPRINT_FILEPATH', os.path.join(mkdtemp(), 'fingerprint'))
+    @patch.object(db_utils, 'sh')
+    def test_load_data_from_s3_fingerprint(self, _mock_sh):
+        """
+        Assuming that the computed db cache file fingerprint is different
+        than the stored fingerprint AND there is a matching fingerprint file
+        in s3, verify that we make a call to load data into the database
+        without running migrations
+        """
+        self.addCleanup(shutil.rmtree, db_utils.CACHE_FOLDER)
+        self.addCleanup(os.remove, db_utils.FINGERPRINT_FILEPATH)
+        _write_temporary_db_cache_files(db_utils.CACHE_FOLDER, database.ALL_DB_FILES)
+
+        # zip the temporary files and push them to a moto s3 bucket
+        expected_fingerprint = 'ccaa8d8dcc7d030cd6a6768db81f90d0ef976c3d'
+        zipfile_name = '{}.tar.gz'.format(expected_fingerprint)
+        zipfile_path = os.path.join(db_utils.CACHE_FOLDER, zipfile_name)
+        with tarfile.open(name=zipfile_path, mode='w:gz') as tar_file:
+            for name in database.ALL_DB_FILES:
+                tar_file.add(os.path.join(db_utils.CACHE_FOLDER, name), arcname=name)
+        key = boto.s3.key.Key(bucket=self.bucket, name=zipfile_name)
+        key.set_contents_from_filename(zipfile_path, replace=False)
+
+        # write the local fingerprint file with a different value than
+        # the computed fingerprint
+        local_fingerprint = '123456789'
+        with open(db_utils.FINGERPRINT_FILEPATH, 'w') as fingerprint_file:
+            fingerprint_file.write(local_fingerprint)
+
+        database.update_local_bokchoy_db_from_s3()
+        calls = [
+            call('{}/scripts/calculate-bokchoy-migrations.sh'.format(Env.REPO_ROOT)),
+            call('{}/scripts/reset-test-db.sh'.format(Env.REPO_ROOT))
+        ]
+        _mock_sh.assert_has_calls(calls)
```

Comment:
The assertions are the same as the test_load_data_from_local_cache testcase.
We should also verify that the files were retrieved from s3 and extracted, and that those were the ones that were used.